crates/gas/src/lib.rs

```diff
@@ -373,56 +411,64 @@ pub trait GasMetering {
         )
     }
 
-    /// Get the gas consumed by the tx
-    fn get_tx_consumed_gas(&self) -> Gas;
-
-    /// Get the gas limit
-    fn get_gas_limit(&self) -> Gas;
-
-    /// Get the protocol gas scale
-    fn get_gas_scale(&self) -> u64;
+    /// Check if the meter ran out of gas. Starts with the initial gas.
+    fn check_limit(&self, gas: Gas) -> Result<()> {
+        self.get_initially_available_gas()
+            .checked_sub(gas)
+            .ok_or_else(|| {
+                Error::TransactionGasExceededError(
+                    self.get_gas_limit()
+                        .get_whole_gas_units(self.get_gas_scale()),
+                )
+            })
+            .and(Ok(()))
+    }
 
-    /// Check if the vps went out of gas. Starts with the gas consumed by the
-    /// transaction.
-    fn check_vps_limit(&self, vps_gas: Gas) -> Result<()> {
-        let total = self
-            .get_tx_consumed_gas()
-            .checked_add(vps_gas)
-            .ok_or(Error::GasOverflow)?;
-        let gas_limit = self.get_gas_limit();
-        if total > gas_limit {
-            return Err(Error::TransactionGasExceededError(
-                gas_limit.get_whole_gas_units(self.get_gas_scale()),
-            ));
-        }
+    /// Add the gas required by a wrapper transaction which is comprised of:
+    ///  - cost of validating the wrapper tx
+    ///  - space that the transaction requires in the block
+    ///  - cost of downloading (as part of the block) the transaction bytes over
+    ///    the network
+    fn add_wrapper_gas(&mut self, tx_bytes: &[u8]) -> Result<()> {
+        self.consume(WRAPPER_TX_VALIDATION_GAS.into())?;
 
-        Ok(())
+        let bytes_len = tx_bytes.len() as u64;
+        self.consume(
+            bytes_len
+                .checked_mul(
+                    STORAGE_OCCUPATION_GAS_PER_BYTE
+                        + NETWORK_TRANSMISSION_GAS_PER_BYTE,
+                )
+                .ok_or(Error::GasOverflow)?
+                .into(),
+        )
     }
 }
 
 /// Gas metering in a transaction
-#[derive(Debug)]
+#[derive(Debug, Default)]
```

Comment:
I don't think it is safe to provide a default implementation for the gas meters given that the `gas_scale` is a protocol parameter